cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -67,11 +67,12 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
         -- The output of @pkg-config --list-all@ also includes a description
         -- for each package, which we do not need.
         let pkgNames = map (takeWhile (not . isSpace)) pkgList
-        (pkgVersions, _errs, exitCode) <-
+        (pkgVersions_, _errs, exitCode) <-
                      getProgramInvocationOutputAndErrors verbosity
                        (programInvocation pkgConfig ("--modversion" : pkgNames))
-        if exitCode == ExitSuccess && length pkgNames == length pkgList
-          then (return . pkgConfigDbFromList . zip pkgNames) (lines pkgVersions)
+        let pkgVersions = lines pkgVersions_
+        if exitCode == ExitSuccess && length pkgVersions == length pkgList
+          then (return . pkgConfigDbFromList . zip pkgNames) pkgVersions
```

Comment:
I think it's slightly clearer to compare the lengths of `pkgVersions` and `pkgNames`, since they are the lists being zipped.

Reply:
Alright, I think it is a safe change since they are identical in length anyway.
(I had chosen pkgList, since pkgNames is derived from it.)

(Here` I kinda wish `getProgramInvocationOutputAndErrors` returned `(status,err,out)` instead.)